hbase-protocol/src/main/java/org/apache/hadoop/hbase/util/ByteStringer.java

```diff
@@ -56,14 +55,16 @@ private ByteStringer() {
    * Wraps a byte array in a {@link ByteString} without copying it.
    */
   public static ByteString wrap(final byte[] array) {
-    return USE_ZEROCOPYBYTESTRING? HBaseZeroCopyByteString.wrap(array): ByteString.copyFrom(array);
+    return USE_ZEROCOPYBYTESTRING
+      ? UnsafeByteOperations.unsafeWrap(array)
+      : ByteString.copyFrom(array);
   }
 
   /**
    * Wraps a subset of a byte array in a {@link ByteString} without copying it.
    */
   public static ByteString wrap(final byte[] array, int offset, int length) {
-    return USE_ZEROCOPYBYTESTRING? HBaseZeroCopyByteString.wrap(array, offset, length):
+    return USE_ZEROCOPYBYTESTRING? UnsafeByteOperations.unsafeWrap(array, offset, length):
       ByteString.copyFrom(array, offset, length);
   }
 }
```

Comment:
What is target for this fix?  What version?

Looking at HBASE-23797, the result of the discussion around CPEPs, the idea is we'd remove hbase-protocol module.

Reply:
HBase 3.0.0. I think it might break someone's setup if it is done in 2.x.

But I agree that PB must be shaded and HBase itself should use only the shaded version. This will make it easier to upgrade in the future and third parties won't be affected.